src/parser/parser.py

```diff
@@ -129,14 +149,18 @@ def _parse_heroes(self, parsed_abilities):
         ).run()
 
         # Ensure it matches the current list of meaningful stats, and raise a warning if not
-        # File diff will also appear in git
-        if not json_utils.compare_json_file_to_dict(
-            self.OUTPUT_DIR + '/json/hero-meaningful-stats.json', parsed_meaningful_stats
-        ):
-            print(
-                'Warning: Non-constant stats have changed. '
+        current_meaningful_stats = json_utils.read(
+            self.OUTPUT_DIR + '/json/hero-meaningful-stats.json'
+        )
+        if current_meaningful_stats != parsed_meaningful_stats:
+            current_keys = set(current_meaningful_stats.keys())
+            new_keys = set(parsed_meaningful_stats.keys())
+            logger.warning(
+                'Non-constant stats have changed. '
                 + "Please update [[Module:HeroData]]'s write_hero_comparison_table "
                 + 'lua function for the [[Hero Comparison]] page.'
+                + f'\nAdded keys: {new_keys - current_keys}'
+                + f'\nRemoved keys: {current_keys - new_keys}'
```

Comment:
With the removal of output-data, this is no longer possible. Instead you could add the wiki json file to your watch list to keep track of any changes

Reply:
Hmm I guess that's true when ran via GHA unfortunately. We will want to look into replacements for this soon. Watch list is not a good system because wiki diffing data structures does not work. i.e. some attributes of hero_wraith changed on line70-72, but the line that says its wraith's attributes are on line60. Since the wiki diff will only show line70-72, we can't see what changed accurately.

Also, recall that this file is not uploaded to the wiki, its only purpose is to more accurately maintain what stats are variable and which ones are constants for all heroes. To fix deadbot not erroring, I added a small check to ensure the path exists before comparing. For now, we can keep this check even though it only works locally.

We can discuss ideas for the replacement later and elsewhere, but if you really want to leave out reading from the wiki from this repo then we can either build it into the deadlock-data repo or make a 3rd repo, as I have lots of ideas to improve the workflow that involve reading from the wiki.

Reply:
Ah it isn't uploaded you're right. I'm not sure the best way to handle that tbh, as even with a successful comparison, we need something more than a log message buried in the gha. But uploading to the wiki is one option no? If there is a change to `meaningful-data.json`, you would see that in your watch list